src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Patch/ConditionalPatchResourceHandler.cs

```diff
@@ -39,19 +42,27 @@ public sealed class ConditionalPatchResourceHandler : ConditionalResourceHandler
             EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
             EnsureArg.IsNotNull(mediator, nameof(mediator));
 
-            _patchService = new JsonPatchService(modelInfoProvider);
+            dynamic service;
```

Comment:
@Ivanidzo4ka @mikaelweave I templated away both the ResourceRequest and ConditionalPatchResourceRequest. Ideally we could guard the TData type to be either Parameters or JsonPatchDocument but I foresee defering type check logic somewhere else. I really hate this code for picking the service but im not sure a full factory pattern is necessary. My other idea would be to lazy an instance of the JsonPatchService and FhirPatchService.

Would love some more input here

Reply:
@mattpasco-microsoft this works but the resource handlers need to be registered explicitly with the DI container due to MediatR and possibly even .NET. [See here for details](https://github.com/jbogard/MediatR/issues/674).

I added the registration for the Patch handlers and used the container for`JsonPatchService` and `FhirParametersPatchService`. I added them where it made sense to me but should be reviewed by someone for familiar with the repo (@Ivanidzo4ka and/or @brendankowitz).